pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1151,34 +872,36 @@ protected IdealState ensureAllPartitionsConsuming(final TableConfig tableConfig,
           // 1. all replicas OFFLINE and metadata IN_PROGRESS/DONE - a segment marked itself OFFLINE during consumption for some reason
           // 2. all replicas ONLINE and metadata DONE - Resolved in https://github.com/linkedin/pinot/pull/2890
           // 3. we should never end up with some replicas ONLINE and some OFFLINE.
-          if (isAllInstancesInState(instanceStateMap, PinotHelixSegmentOnlineOfflineStateModelGenerator.OFFLINE_STATE)
-              || !(isTooSoonToCorrect(tableNameWithType, segmentId, now))) {
-
-            // No instances are consuming, so create a new consuming segment.
-            LLCSegmentName newLLCSegmentName = makeNextLLCSegmentName(segmentName, partition, now);
-            LOGGER.info("Creating CONSUMING segment {} for {} partition {}", newLLCSegmentName.getSegmentName(),
-                tableNameWithType, partition);
-
-            // To begin with, set startOffset to the oldest available offset in the stream. Fix it to be the one we want,
-            // depending on what the prev segment had.
-            long startOffset = getPartitionOffset(streamConfig, OffsetCriteria.SMALLEST_OFFSET_CRITERIA, partition);
-            LOGGER.info("Found smallest offset {} for table {} for partition {}", startOffset, tableNameWithType,
```

Comment:
ALso for this one. Can we somewhere log if we are adjusting the offset? (even without data loss)

Reply:
Also logged in `createNewSegmentZKMetadata()` lol